### PR TITLE
feat(provider/google): Add ILB listeners.

### PR DESCRIPTION
**Pull request by jtk54**
Adds UI for adding ILB listeners and updating ports on existing listeners. Depends on https://github.com/spinnaker/clouddriver/pull/1693.

### CODE REVIEW

File: app/scripts/modules/google/loadBalancer/configure/internal/listeners/listeners.html [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<ng-form name="listeners">
+  <gce-internal-load-balancer-listener load-balancer="ctrl.loadBalancer" subnet-options="ctrl.subnetOptions"
+                delete-listener="ctrl.removeListener($index)"
```
Comment (danielpeach) [inline app/scripts/modules/google/loadBalancer/configure/internal/listeners/listeners.html]: These attributes should be aligned with `load-balancer`.

File: app/scripts/modules/google/src/loadBalancer/configure/internal/gceCreateInternalLoadBalancer.controller.ts [hunk 2]
```diff
@@ -167,22 +174,25 @@ class InternalLoadBalancerCtrl extends CommonGceLoadBalancerCtrl implements ng.I
...
     if (!this.subnetOptions.includes(this.loadBalancer.subnet)) {
       this.loadBalancer.subnet = this.subnetOptions[0];
     }
   }
 
+  private updateSubnetOptions(): void {
+    this.subnetOptions = this.subnets
+        .filter((subnet) => {
+          return subnet.region === this.loadBalancer.region &&
+              (subnet.account === this.loadBalancer.credentials || subnet.account === this.loadBalancer.account) &&
```
Comment (danielpeach) [inline app/scripts/modules/google/src/loadBalancer/configure/internal/gceCreateInternalLoadBalancer.controller.ts]: Indentation is off here.

File: app/scripts/modules/google/src/loadBalancer/configure/internal/gceCreateInternalLoadBalancer.controller.ts [hunk 3]
```diff
@@ -167,22 +174,25 @@ class InternalLoadBalancerCtrl extends CommonGceLoadBalancerCtrl implements ng.I
...
-
+    this.updateSubnetOptions();
     if (!this.subnetOptions.includes(this.loadBalancer.subnet)) {
       this.loadBalancer.subnet = this.subnetOptions[0];
     }
   }
 
+  private updateSubnetOptions(): void {
+    this.subnetOptions = this.subnets
+        .filter((subnet) => {
```
Comment (danielpeach) [inline app/scripts/modules/google/src/loadBalancer/configure/internal/gceCreateInternalLoadBalancer.controller.ts]: Remove parens around function parameter here and below.

File: app/scripts/modules/google/src/loadBalancer/configure/internal/gceCreateInternalLoadBalancer.controller.ts [hunk 4]
```diff
@@ -218,29 +228,41 @@ class InternalLoadBalancerCtrl extends CommonGceLoadBalancerCtrl implements ng.I
     this.loadBalancer.loadBalancerName = this.getName(this.loadBalancer, this.application);
   }
 
-  public setSessionAffinity (viewState: ViewState): void {
+  public addListener (): void {
```
Comment (danielpeach) [inline app/scripts/modules/google/src/loadBalancer/configure/internal/gceCreateInternalLoadBalancer.controller.ts]: Spacing between method name and `()` is inconsistent.

File: app/scripts/modules/google/src/loadBalancer/loadBalancer.setTransformer.ts [hunk 5]
```diff
@@ -15,14 +16,32 @@ export class GceLoadBalancerSetTransformer {
...
   }
 
+  private static normalizeInternalLoadBalancerGroup(group: InternalLoadBalancer[]): InternalLoadBalancer {
+    const normalized = cloneDeep(group[0]);
+
+    normalized.listeners = group.map((loadBalancer) => {
+      return {
+        ports: loadBalancer.ports,
+        name: loadBalancer.name,
+        // ipAddress: loadBalancer.ipAddress,
```
Comment (danielpeach) [inline app/scripts/modules/google/src/loadBalancer/loadBalancer.setTransformer.ts]: Remove comment.

File: app/scripts/modules/google/src/loadBalancer/details/loadBalancerDetail.controller.js [hunk 6]
```diff
@@ -118,6 +118,29 @@ module.exports = angular.module('spinnaker.loadBalancer.gce.details.controller',
             return [representativeLb];
           });
 
+      } else if (($scope.loadBalancer.provider === 'gce' || $scope.loadBalancer.type === 'gce') && $scope.loadBalancer.loadBalancerType === 'INTERNAL') {
+        var detailsPromises = $scope.loadBalancer.listeners.map((listener) => {
```
Comment (danielpeach) [inline app/scripts/modules/google/src/loadBalancer/details/loadBalancerDetail.controller.js]: Looks like `detailsPromises` is defined up above. `var` isn't block scoped in JS, so switch to `const`.

File: app/scripts/modules/google/loadBalancer/configure/internal/internalLoadBalancer.write.service.js [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+const angular = require('angular');
+
+import { INFRASTRUCTURE_CACHE_SERVICE, TASK_EXECUTOR } from '@spinnaker/core';
+
+module.exports = angular.module('spinnaker.deck.gce.internalLoadBalancer.write.service', [
+  TASK_EXECUTOR,
+  INFRASTRUCTURE_CACHE_SERVICE
+])
+  .factory('gceInternalLoadBalancerWriter', function (taskExecutor, infrastructureCaches) {
```
Comment (jtk54) [inline app/scripts/modules/google/loadBalancer/configure/internal/internalLoadBalancer.write.service.js]: Yep those turned out remarkably similar. I can refactor those into something more general like a `multipleListenerLoadBalancer.write.service`.

File: app/scripts/modules/google/src/loadBalancer/configure/internal/gceCreateInternalLoadBalancer.controller.ts (hunk 4 above)
Comment (jtk54) [inline app/scripts/modules/google/src/loadBalancer/configure/internal/gceCreateInternalLoadBalancer.controller.ts]: Made these all spaces.

**Comment by jtk54** [comment]
Addressed style comments and did the small writer service refactor, re-tested with ILBs and L7 for sanity, everything seems to be fine.

**Comment by jtk54** [comment]
Closes https://github.com/spinnaker/spinnaker/issues/1717.

**Pull request merged**